pkg/scalers/azure_pipelines_scaler.go

```diff
@@ -306,13 +329,28 @@ func validatePoolID(ctx context.Context, poolID string, metadata *azurePipelines
 	return result.ID, nil
 }
 
-func getAzurePipelineRequest(ctx context.Context, urlString string, metadata *azurePipelinesMetadata, httpClient *http.Client) ([]byte, error) {
+func getAzurePipelineRequest(ctx context.Context, logger logr.Logger, urlString string, metadata *azurePipelinesMetadata, podIdentity kedav1alpha1.AuthPodIdentity, httpClient *http.Client) ([]byte, error) {
 	req, err := http.NewRequestWithContext(ctx, "GET", urlString, nil)
 	if err != nil {
 		return []byte{}, err
 	}
 
-	req.SetBasicAuth("", metadata.personalAccessToken)
+	switch podIdentity.Provider {
+	case "", kedav1alpha1.PodIdentityProviderNone:
+		//PAT
+		logger.V(1).Info("making request to ADO REST API using PAT")
+		req.SetBasicAuth("", metadata.personalAccessToken)
+	case kedav1alpha1.PodIdentityProviderAzureWorkload:
+		//ADO Resource token
+		resource := "499b84ac-1321-427f-aa17-267ca6975798"
+		logger.V(1).Info("making request to ADO REST API using managed identity")
+		aadToken, err := azure.GetAzureADWorkloadIdentityToken(ctx, "", resource)
+		if err != nil {
+			return []byte{}, fmt.Errorf("cannot create workload identity credentials: %w", err)
+		}
+		logger.V(1).Info("token acquired setting auth header as 'bearer XXXXXX'")
+		req.Header.Set("Authorization", "Bearer "+aadToken.AccessToken)
```

Comment:
Please, instead of doing this, use the new approach using azidentity as it's described here: https://github.com/microsoft/azure-devops-go-api/issues/142#issuecomment-1853276650

You can access to it using `azure.NewChainedCredential(` and providing the required info. For example:
https://github.com/kedacore/keda/blob/9c23daee8a0c84f98b641df3391b5276ca6653ec/pkg/scalers/azure/azure_managed_prometheus_http_round_tripper.go#L40-L43
returned credentials has a function `GetToken` that you can call, providing the scopes that you want to request (using azidentity the scope is `"499b84ac-1321-427f-aa17-267ca6975798/.default"`), something like 
```golang
creds.GetToken(ctx, policy.TokenRequestOptions{
    Scopes: []string{
        "499b84ac-1321-427f-aa17-267ca6975798/.default",
    },
})
```

Reply:
yes you're right this is better approach. I will update token auth this week

Reply:
No worries, and sorry for the huge delay on reviewing this PR, we had to migrate to the new subscription and it's been a pain :( I want to apologize for that

Reply:
no problem @JorTurFer we were able to run this using our own build. 

Regarding the changes to obtain token. Unfortunately token (auth context) is required before scaler is created. Metadata is only one place shared with all required function. I've updated metadata with authContext field but it is not actual metadata so I want to know your opinion on this. THX